runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6267,9 +6397,74 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
             *aotCachedMethod = javaVM->sharedClassConfig->findCompiledMethodEx1(vmThread, J9_ROM_METHOD_FROM_RAM_METHOD(method), &flags);
             if (!(flags & J9SHR_AOT_METHOD_FLAG_INVALIDATED))
                {
+#ifdef COMPRESS_AOT_DATA
+
+
+               TR_AOTMethodHeader *aotMethodHeader = (TR_AOTMethodHeader*)(((J9JITDataCacheHeader*)*aotCachedMethod) + 1);
+               if (aotMethodHeader->flags & TR_AOTMethodHeader_CompressedMethodInCache)
+                  {
+                  /**
+                    * For each AOT compiled method we store in the shareclass cache we have follwing layout.
+                    * ---------------------------------------------------------------------------------------
+                    * | CompiledMethodWrapper | J9JITDataCacheHeader | AOTMethodHeader | Metadata | Codedata|
+                    * ---------------------------------------------------------------------------------------
+                    * When we compile and store a method, it stores J9RomMethod for the method in the CompiledMethodWrapper along side
+                    * size of metadata and codedata in the cache. Now when we request a method from shared class cache
+                    * it returns address that points to J9JITDataCacheHeader. As while decompressing the data, we need information about
+                    * the size of original data (We get that information from AOTMethodHeader) and size of deflated data in the cache (Which is stored in
+                    * CompiledMethodWrapper)
+                    * So to access data size in shared class cache, we need to access CompiledMethodWrapper.
+                    */
+                  int sizeOfCompressedDataInCache = *((uint32_t*)(*aotCachedMethod)-2);
```

Comment:
@dsouzai  I have to revert back this in https://github.com/eclipse/openj9/pull/5926/commits/60ef6542ebddb980d1e8f2807b63fe65194fd6c0 
Turns out when we store a compiled method in the shared class cache, it adds another header before the metadata which contains J9RomMethod (For quick comparison in the cache for look-up and the size data occupied for that method in the cache. J9JITDataCacheHeader does not contains the information we need.

Reply:
Here is how method in shared class cache looks like.

| CompiledMethodWrapper | J9JITDataCacheHeader | AOTMethodHeader | Metadata | Codedata|
|---------|---------|-----------|--------------|------------|

When JIT sends data to store in shared class cache,  it sends address pointing at J9JITDataCacheHeader while CompiledMethodWrapper is added in SharedClassCache API.

Reply:
Size stored in `J9JITDataCacheHeader` is not `sizeOfCompressedDataInCache` which is what I need while inflating data.

Reply:
Can't we access it via `CompiledMethodWrapper` then, ie:

```
CompiledMethodWrapper *wrapper = (CompiledMethodWrapper *)(*aotCachedMethod);
int sizeOfCompressedDataInCache = wrapper->dataLength;
```

Reply:
Added in https://github.com/eclipse/openj9/pull/5926/commits/79d9f4618ef87643cd4ffe77314559d307bacac2

Reply:
Looks good, thanks